lua/Connector.lua

```diff
@@ -65,7 +66,41 @@ function writebyte(addr, value, domain)
   end
 end
 
-local socket = require("socket.core")
+function get_lua_version()
+    local major, minor = _VERSION:match("Lua (%d+)%.(%d+)")
+    assert(tonumber(major) == 5)
+    if tonumber(minor) >= 4 then
+        return "5-4"
+    end
+    return "5-1"
+end
+
+function get_os()
+    local the_os, ext, arch
+    if package.config:sub(1,1) == "\\" then
+        the_os, ext = "windows", "dll"
+        arch = os.getenv"PROCESSOR_ARCHITECTURE"
+    else
+        -- TODO: macos?
+        the_os, ext = "linux", "so"
+        return the_os, ext, "" -- linux doesn't use/have the arch
+    end
+
+    if (arch or ""):match"64" then
+        arch = "/x64"
+    else
+        arch = "/x32"

```

Comment:
* using find (in case we ever need lua 5.0)
* remove leading `/` (added below) to be less confusing
* fixed wrong return value for 32bit
```suggestion
    if arch:find("64") ~= nil then
        arch = "x64"
    else
        arch = "x86"
```